api/devicerepository.proto

```diff
@@ -308,6 +312,20 @@ message MessagePayloadFormatter {
   PayloadFormatter formatter = 1 [(validate.rules).enum.defined_only = true];
   // Parameter for the formatter, must be set together.
   string formatter_parameter = 2;
+
+  string codec_id = 3 [(gogoproto.customname) = "CodecID", (validate.rules).string = {pattern: "^([a-z0-9](?:[-]?[a-z0-9]){2,}|)?$" , max_len: 36}];
+
+  message Example {
+    string description = 1;
+    message Input {
+      uint32 f_port = 1 [(validate.rules).uint32.lte = 255];
+      bytes frm_payload = 2 [(gogoproto.customname) = "FRMPayload"];
+    };
+    Input input = 2;
+    google.protobuf.Struct output = 3;
```

Comment:
Why is the input following the input parameters, but the output is not? I.e. where are the warnings and errors?

Reply:
Changed to using a Struct for the input as well.

Reply:
TBH, I think this becomes a bit nasty. We know the input and output parameters here so they can be typed. We just need to differentiate between encoding and decoding. Why do you use `Struct` here?